### ISSUE TITLE
Pycaret model deploy to aws subfolder

### ISSUE THREAD

**Issue by sagarnildass** [pycaret/pycaret#1141]
Hi

Is there any method where I can the deploy_model function of pycaret to deploy a model in a specific subfolder within a AWS bucket? Right now, it only seems, that within a bucket's first level, can a model be deployed.

**Comment by moezali1** [comment]
@sagarnildass Yes. It's not possible to deploy into a sub-folder. Will you like to work on this? It should be a simple change.

**Comment by moezali1** [comment]
@amjadraza Can you help with this issue?

**Comment by amjadraza** [comment]
Let me review it. Looks like a simple change.

**Comment by amjadraza** [comment]
@moezali1 assign this issue to me please.

**Comment by moezali1** [comment]
@amjadraza Thank you so much. Assigned to you. 

I am not sure how should we handle the folder names for Azure/AWS/GCP but maybe some logic to parse the `model_name` parameter in the `deploy_model` function. 

I will leave it up to you to decide the most optimal way to implement this use case. Let me know if you feel like catching the brain!

**Comment by moezali1** [comment]
@sagarnildass Feel free to chip in.

**Comment by wkuopt** [comment]
This seems to be the same issue as #1443. Please take a look at the PR and see if that makes sense to you :)

**Comment by moezali1** [comment]
Closing as part of the 4.0 backlog cleanup — see #4177.

PyCaret 4.0 is a ground-up rewrite (OOP-only API, sklearn-native engine, Python 3.11+, sklearn ≥ 1.7) and almost all pre-4.0 issues either reference removed APIs (`setup()`, `plot_model`, `evaluate_model`, `interpret_model`, `automl`, the legacy time-series module, the Streamlit dashboard, …) or no longer reproduce on the new code path.

**If this still affects you in 4.0, please re-file as a new issue** with:

- the experiment constructor call (`Experiment(...)`),
- the operation that fails,
- the full traceback,
- your Python / sklearn / pycaret versions.

Resources for 4.0:

- Docs: https://pycaret.org/docs
- Migrate from 3.x: https://pycaret.org/docs/guides/migrating-from-3-x
- API reference: https://pycaret.org/reference
- FAQ: https://pycaret.org/docs/resources/faqs

Thanks for filing — and apologies for the long latency. The repo had drifted; we're starting clean.
